### PR TITLE
Reduce excessive number of spawned synclist tasks

### PR DESCRIPTION
**Pull request by cutwater**
Fix large number of enqueued synclist related tasks by doing the
following changes:
  * Call `add_and_remove` tasks directly.
  * Call `curate_synclist_repository` task in batches.
  * Batch size is set to 200.

### CODE REVIEW

File: galaxy_ng/app/api/ui/viewsets/my_synclist.py [hunk 1]
```diff
@@ -35,13 +35,12 @@ def get_queryset(self):
         )
 
     @action(detail=True, methods=["post"])
-    def curate(self, request, pk=None):
-        locks = [pk]
-        task_args = (pk,)
-        task_kwargs = {}
-
+    def curate(self, request, pk):
+        synclist = models.SyncList.objects.get(pk)
```
Comment (newswangerd) [inline galaxy_ng/app/api/ui/viewsets/my_synclist.py:39]: maybe `get_object_or_404` or something similar

File: galaxy_ng/app/tasks/synclist.py [hunk 2]
```diff
@@ -20,7 +19,8 @@
 from galaxy_ng.app import models
 
 log = logging.getLogger(__name__)
-pf = pprint.pformat
+
+SYNCLIST_BATCH_SIZE = 200
```
Comment (newswangerd) [inline galaxy_ng/app/tasks/synclist.py:23]: should this be configurable in settings.py? That way we can tweak it without a code deployment

**Comment by bmbouter** [comment]
From a high level this contains all the aspects I was hoping for both batching via the introduction of an intermediary task, and the removal of the extra `enqueue_with_reservation`. :+1:

**Review by alikins — APPROVED** [review]
Seems okay to me. It reduces the granularity of the tasks, but that is the intention. 
Only concern is it is has any impact on error handling. In particular, if errors raised from curate_synclist_repository() lose any context. I haven't tested it, and can't point to any
cases where it could cause a problem, so assuming it should be okay.

**Review by newswangerd — APPROVED** [review]
LGTM

File: galaxy_ng/app/tasks/synclist.py [hunk 3]
```diff
@@ -94,6 +95,11 @@ def curate_all_synclist_repository(upstream_repository_name, **kwargs):
     task_group.finish()
 
 
+def curate_synclist_repository_batch(synclist_pks, **kwargs):
+    for synclist_pk in synclist_pks:
+        curate_synclist_repository(synclist_pk=synclist_pk)
```
[the remark this answers is not in the document]
Comment (alikins) [reply galaxy_ng/app/tasks/synclist.py:100]: So a curate_synclist_repository_batch() will end up running entirely on one worker?

Will it block if any curate_synclist_repository() call blocks?

**Pull request merged**